pubsub/aws/snssqs/snssqs.go

```diff
@@ -431,25 +459,61 @@ func (s *snsSqs) createQueue(queueName string) (*sqsQueueInfo, error) {
 	}, nil
 }
 
-func (s *snsSqs) getOrCreateQueue(queueName string) (*sqsQueueInfo, error) {
-	queueArn, ok := s.queues[queueName]
+func (s *snsSqs) getQueueArn(queueName string) (*sqsQueueInfo, error) {
+	queueURLOutput, err := s.sqsClient.GetQueueUrl(&sqs.GetQueueUrlInput{QueueName: aws.String(queueName), QueueOwnerAWSAccountId: aws.String(s.metadata.accountID)})
+	if err != nil {
+		return nil, fmt.Errorf("error: %w while getting url of queue: %s", err, queueName)
+	}
+	url := queueURLOutput.QueueUrl
+
+	var getQueueOutput *sqs.GetQueueAttributesOutput
+	getQueueOutput, err = s.sqsClient.GetQueueAttributes(&sqs.GetQueueAttributesInput{QueueUrl: url, AttributeNames: []*string{aws.String("QueueArn")}})
+	if err != nil {
+		return nil, fmt.Errorf("error: %w while getting information for queue: %s, with url: %s", err, queueName, *url)
+	}
 
+	return &sqsQueueInfo{arn: *getQueueOutput.Attributes["QueueArn"], url: *url}, nil
+}
+
+func (s *snsSqs) getOrCreateQueue(queueName string) (*sqsQueueInfo, error) {
+	var (
+		err       error
+		queueInfo *sqsQueueInfo
+		ok        bool
+	)
+
+	s.queuesLock.RLock()
+	queueInfo, ok = s.queues[queueName]
+	s.queuesLock.RUnlock()
 	if ok {
-		s.logger.Debugf("Found queue arn for %s: %s", queueName, queueArn)
+		s.logger.Debugf("Found queue arn for %s: %s", queueName, queueInfo.arn)
 
-		return queueArn, nil
+		return queueInfo, nil
 	}
 	// creating queues is idempotent, the names serve as unique keys among a given region.
 	s.logger.Debugf("No queue arn found for %s\nCreating queue", queueName)
 
-	queueInfo, err := s.createQueue(queueName)
-	if err != nil {
-		s.logger.Errorf("Error creating queue %s: %v", queueName, err)
+	sanitizedName := nameToAWSSanitizedName(queueName, s.metadata.fifo)
 
-		return nil, err
+	if !s.metadata.disableEntityManagement {
+		queueInfo, err = s.createQueue(sanitizedName)
+		if err != nil {
+			s.logger.Errorf("Error creating queue %s: %v", queueName, err)
+
+			return nil, err
+		}
+	} else {
+		queueInfo, err = s.getQueueArn(sanitizedName)
+		if err != nil {
+			s.logger.Errorf("error fetching info for queue %s: %w", queueName, err)
+
+			return nil, err
+		}
 	}
 
+	s.queuesLock.Lock()
 	s.queues[queueName] = queueInfo
```

Comment:
How about use `sync.Map`?

Reply:
Good option. I just followed the implementation of https://github.com/dapr/components-contrib/blob/cac03dbe874f6aee9148dc3002b734b8ea5d9c4f/pubsub/azure/servicebus/servicebus.go#L88 but I prefer doing things better